middleware_auth.go

```diff
@@ -265,26 +265,34 @@ func RequiresPermission(opts RequiresPermissionMiddlewareOpts) Middleware {
 }
 
 func (rp *requiresPermissionHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	ctx := r.Context()
 	vars := GetVars(r)
-	var resource string
+	resources := []string{}
 	var status int
 	var err error
 	if rp.opts.ResourceFunc != nil {
-		resource, status, err = rp.opts.ResourceFunc(r)
+		resources, status, err = rp.opts.ResourceFunc(r)
 		if err != nil {
 			http.Error(rw, err.Error(), status)
 			return
 		}
 	} else {
 		for _, level := range rp.opts.ResourceLevels {
 			if resourceVal, exists := vars[level]; exists {
-				resource = resourceVal
+				resources = []string{resourceVal}
 				break
 			}
 		}
 	}
 
+	for _, item := range resources {
+		if ok := rp.handleResource(rw, r, next, item); !ok {
+			return
+		}
+	}
+}
+
+func (rp *requiresPermissionHandler) handleResource(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc, resource string) bool {
```

Comment:
This will let the request through if just 1 of the resources passes the permission check since you're calling the next function. I would change this to just do the check and let the function above handle all the HTTP stuff